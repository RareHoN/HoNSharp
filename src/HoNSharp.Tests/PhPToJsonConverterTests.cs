using ApprovalTests;
using ApprovalTests.Reporters;
using HoNSharp.Converters;
using Newtonsoft.Json;

namespace HoNSharp.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class PhPToJsonConverterTests
    {
        private readonly PhPToJsonConverter _converter;

        public PhPToJsonConverterTests()
        {
            _converter = new PhPToJsonConverter();
        }

        [Fact]
        public void ConvertToJson_GetAccountAllHeroStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_account_all_hero_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_WithDateTimeOffsetError_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_datetimeoffset_error_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_WithUnexpectedCharacterError_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_unexpected_character_error_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_MatchHistoryOverviewMidwars_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("match_history_overview_midwars_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_MatchHistoryOverviewNormal_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("match_history_overview_normal_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsCampaign_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_campaign_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsCampaignMidwars_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_campaign_midwars_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsMastery_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_mastery_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_SimpleStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("simple_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        private string ReadFile(string responseFileName)
        {
            var responseFilePath = Path.Combine(Directory.GetCurrentDirectory(), "TestData", responseFileName);

            return File.ReadAllText(responseFilePath);
        }

        private string FormatJson(string json)
        {
            var deserialized = JsonConvert.DeserializeObject(json);
            return JsonConvert.SerializeObject(deserialized, Formatting.Indented);
        }
    }
}