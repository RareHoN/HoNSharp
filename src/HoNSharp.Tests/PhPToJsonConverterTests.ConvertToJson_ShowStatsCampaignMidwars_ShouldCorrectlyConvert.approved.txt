{
  "super_id": 83079,
  "nickname": "RareHoN",
  "standing": 3,
  "account_type": 4,
  "account_id": 83079,
  "level": 1,
  "level_exp": 0,
  "discos": 0,
  "possible_discos": 0,
  "games_played": 3,
  "num_bot_games_won": 0,
  "total_games_played": 3,
  "total_discos": 0,
  "acc_secs": 3117,
  "acc_games_played": 3,
  "acc_discos": 0,
  "rnk_secs": 0,
  "rnk_games_played": 0,
  "rnk_discos": 0,
  "cs_secs": 0,
  "cs_games_played": 0,
  "cs_discos": 0,
  "mid_games_played": 0,
  "mid_discos": 0,
  "rift_games_played": 0,
  "rift_discos": 0,
  "last_activity": "11/5/2022",
  "create_date": "11/5/2022",
  "season": 22,
  "cam_games_played": 3,
  "cam_wins": 2,
  "cam_losses": 1,
  "cam_concedes": 1,
  "cam_concedevotes": 2,
  "cam_buybacks": 0,
  "cam_discos": 0,
  "cam_kicked": 0,
  "cam_amm_solo_rating": {
    "MinValue": {
      "MinValue": null,
      "Epsilon": {
        "MinValue": null,
        "Epsilon": null,
        "MaxValue": {
          "MinValue": null,
          "Epsilon": null,
          "MaxValue": null,
          "PositiveInfinity": {
            "MinValue": null,
            "Epsilon": null,
            "MaxValue": null,
            "PositiveInfinity": null,
            "NegativeInfinity": {
              "MinValue": null,
              "Epsilon": null,
              "MaxValue": null,
              "PositiveInfinity": null,
              "NegativeInfinity": null,
              "NaN": {
                "MinValue": null,
                "Epsilon": null,
                "MaxValue": null,
                "PositiveInfinity": null,
                "NegativeInfinity": null,
                "NaN": null
              }
            },
            "NaN": null
          },
          "NegativeInfinity": null,
          "NaN": null
        },
        "PositiveInfinity": null,
        "NegativeInfinity": null,
        "NaN": null
      },
      "MaxValue": null,
      "PositiveInfinity": null,
      "NegativeInfinity": null,
      "NaN": null
    },
    "Epsilon": null,
    "MaxValue": null,
    "PositiveInfinity": null,
    "NegativeInfinity": null,
    "NaN": null
  },
  "cam_amm_solo_count": 0,
  "cam_amm_solo_conf": 0,
  "cam_amm_solo_prov": 0,
  "cam_amm_solo_pset": 0,
  "cam_amm_team_rating": null,
  "cam_amm_team_count": 3,
  "cam_amm_team_conf": 0,
  "cam_amm_team_prov": 0,
  "cam_amm_team_pset": 0,
  "cam_herokills": 33,
  "cam_herodmg": 78289,
  "cam_heroexp": 17079,
  "cam_herokillsgold": 25563,
  "cam_heroassists": 73,
  "cam_deaths": 17,
  "cam_goldlost2death": 0,
  "cam_secs_dead": 126,
  "cam_teamcreepkills": 195,
  "cam_teamcreepdmg": 112642,
  "cam_teamcreepexp": 6308,
  "cam_teamcreepgold": 8614,
  "cam_neutralcreepkills": 9,
  "cam_neutralcreepdmg": 7960,
  "cam_neutralcreepexp": 1509,
  "cam_neutralcreepgold": 701,
  "cam_bdmg": 6766,
  "cam_bdmgexp": 0,
  "cam_razed": 2,
  "cam_bgold": 1210,
  "cam_denies": 9,
  "cam_exp_denied": 386,
  "cam_gold": 44574,
  "cam_gold_spent": 44000,
  "cam_exp": 31311,
  "cam_actions": 9227,
  "cam_secs": 3117,
  "cam_consumables": 27,
  "cam_wards": 4,
  "cam_em_played": 0,
  "cam_level": 1,
  "cam_level_exp": 0,
  "cam_min_exp": 0,
  "cam_max_exp": 0,
  "cam_time_earning_exp": 3117,
  "cam_bloodlust": 0,
  "cam_doublekill": 9,
  "cam_triplekill": 2,
  "cam_quadkill": 1,
  "cam_annihilation": 0,
  "cam_ks3": 3,
  "cam_ks4": 3,
  "cam_ks5": 1,
  "cam_ks6": 1,
  "cam_ks7": 0,
  "cam_ks8": 0,
  "cam_ks9": 0,
  "cam_ks10": 0,
  "cam_ks15": 0,
  "cam_smackdown": 0,
  "cam_humiliation": 0,
  "cam_nemesis": 59,
  "cam_retribution": 1,
  "acc_pub_skill": 1564,
  "acc_wins": 2,
  "acc_losses": 1,
  "acc_concedes": 1,
  "acc_concedevotes": 2,
  "acc_buybacks": 0,
  "acc_kicked": 0,
  "acc_amm_solo_rating": null,
  "acc_amm_solo_count": 0,
  "acc_amm_solo_conf": 0,
  "acc_amm_solo_prov": 0,
  "acc_amm_solo_pset": 0,
  "acc_amm_team_rating": null,
  "acc_amm_team_count": 3,
  "acc_amm_team_conf": 0,
  "acc_amm_team_prov": 0,
  "acc_amm_team_pset": 0,
  "acc_herokills": 33,
  "acc_herodmg": 78289,
  "acc_heroexp": 17079,
  "acc_herokillsgold": 25563,
  "acc_heroassists": 73,
  "acc_deaths": 17,
  "acc_goldlost2death": 0,
  "acc_secs_dead": 126,
  "acc_teamcreepkills": 195,
  "acc_teamcreepdmg": 112642,
  "acc_teamcreepexp": 6308,
  "acc_teamcreepgold": 8614,
  "acc_neutralcreepkills": 9,
  "acc_neutralcreepdmg": 7960,
  "acc_neutralcreepexp": 1509,
  "acc_neutralcreepgold": 701,
  "acc_bdmg": 6766,
  "acc_bdmgexp": 0,
  "acc_razed": 2,
  "acc_bgold": 1210,
  "acc_denies": 9,
  "acc_exp_denied": 386,
  "acc_gold": 44574,
  "acc_gold_spent": 44000,
  "acc_exp": 31311,
  "acc_actions": 9227,
  "acc_consumables": 27,
  "acc_wards": 4,
  "acc_em_played": 0,
  "acc_level": 1,
  "acc_level_exp": 0,
  "acc_min_exp": 0,
  "acc_max_exp": 0,
  "acc_time_earning_exp": 3117,
  "acc_bloodlust": 0,
  "acc_doublekill": 9,
  "acc_triplekill": 2,
  "acc_quadkill": 1,
  "acc_annihilation": 0,
  "acc_ks3": 3,
  "acc_ks4": 3,
  "acc_ks5": 1,
  "acc_ks6": 1,
  "acc_ks7": 0,
  "acc_ks8": 0,
  "acc_ks9": 0,
  "acc_ks10": 0,
  "acc_ks15": 0,
  "acc_smackdown": 0,
  "acc_humiliation": 0,
  "acc_nemesis": 59,
  "acc_retribution": 1,
  "cam_cs_games_played": 3,
  "cam_cs_wins": 2,
  "cam_cs_losses": 1,
  "cam_cs_concedes": 1,
  "cam_cs_concedevotes": 2,
  "cam_cs_buybacks": 0,
  "cam_cs_discos": 0,
  "cam_cs_kicked": 0,
  "cam_cs_amm_solo_rating": null,
  "cam_cs_amm_solo_count": 0,
  "cam_cs_amm_solo_conf": 0,
  "cam_cs_amm_solo_prov": 0,
  "cam_cs_amm_solo_pset": 0,
  "cam_cs_amm_team_rating": null,
  "cam_cs_amm_team_count": 3,
  "cam_cs_amm_team_conf": 0,
  "cam_cs_amm_team_prov": 0,
  "cam_cs_amm_team_pset": 0,
  "cam_cs_herokills": 33,
  "cam_cs_herodmg": 78289,
  "cam_cs_heroexp": 17079,
  "cam_cs_herokillsgold": 25563,
  "cam_cs_heroassists": 73,
  "cam_cs_deaths": 17,
  "cam_cs_goldlost2death": 0,
  "cam_cs_secam_cs_dead": 126,
  "cam_cs_teamcreepkills": 195,
  "cam_cs_teamcreepdmg": 112642,
  "cam_cs_teamcreepexp": 6308,
  "cam_cs_teamcreepgold": 8614,
  "cam_cs_neutralcreepkills": 9,
  "cam_cs_neutralcreepdmg": 7960,
  "cam_cs_neutralcreepexp": 1509,
  "cam_cs_neutralcreepgold": 701,
  "cam_cs_bdmg": 6766,
  "cam_cs_bdmgexp": 0,
  "cam_cs_razed": 2,
  "cam_cs_bgold": 1210,
  "cam_cs_denies": 9,
  "cam_cs_exp_denied": 386,
  "cam_cs_gold": 44574,
  "cam_cs_gold_spent": 44000,
  "cam_cs_exp": 31311,
  "cam_cs_actions": 9227,
  "cam_cs_secs": 3117,
  "cam_cs_consumables": 27,
  "cam_cs_wards": 4,
  "cam_cs_em_played": 0,
  "cam_cs_level": 1,
  "cam_cs_level_exp": 0,
  "cam_cs_min_exp": 0,
  "cam_cs_max_exp": 0,
  "cam_cs_time_earning_exp": 3117,
  "cam_cs_bloodlust": 0,
  "cam_cs_doublekill": 9,
  "cam_cs_triplekill": 2,
  "cam_cs_quadkill": 1,
  "cam_cs_annihilation": 0,
  "cam_cs_ks3": 3,
  "cam_cs_ks4": 3,
  "cam_cs_ks5": 1,
  "cam_cs_ks6": 1,
  "cam_cs_ks7": 0,
  "cam_cs_ks8": 0,
  "cam_cs_ks9": 0,
  "cam_cs_ks10": 0,
  "cam_cs_ks15": 0,
  "cam_cs_smackdown": 0,
  "cam_cs_humiliation": 0,
  "cam_cs_nemesis": 59,
  "cam_cs_retribution": 1,
  "name": null,
  "rank": "None",
  "favHero1": "xalynx",
  "favHero1Time": 33.33333206176758,
  "favHero1_2": "Hero_Xalynx",
  "favHero2": "flint_beastwood",
  "favHero2Time": 33.33333206176758,
  "favHero2_2": "Hero_FlintBeastwood",
  "favHero3": "forsaken_archer",
  "favHero3Time": 33.33333206176758,
  "favHero3_2": "Hero_ForsakenArcher",
  "favHero4": null,
  "favHero4Time": 0,
  "favHero4_2": null,
  "favHero5": null,
  "favHero5Time": 0,
  "favHero5_2": null,
  "dice_tokens": 1,
  "season_level": 0,
  "slot_id": 5,
  "my_upgrades": {
    "0": "m.allmodes.pass",
    "1": "h.AllHeroes.Hero",
    "2": "m.Super-Taunt"
  },
  "selected_upgrades": {
    "0": "ai.Eye Heart HoN",
    "1": "cc.championofnewerth",
    "2": "cs.Epic Mastery Badge"
  },
  "game_tokens": 0,
  "my_upgrades_info": {},
  "creep_level": 0,
  "timestamp": 1668279753,
  "matchIds": "000513867 000513922 000513980",
  "matchDates": "11/08/202211/08/202211/08/2022",
  "k_d_a": "11/5/24",
  "avgGameLength": 1039,
  "avgXP_min": 602.7141723632812,
  "avgDenies": 3,
  "avgCreepKills": 65,
  "avgNeutralKills": 3,
  "avgActions_min": 177.6130828857422,
  "avgWardsUsed": 1.3333333730697632,
  "quest_stats": {
    "error": {
      "quest_status": 0,
      "leaderboard_status": 0
    }
  },
  "highest_level_current": 0,
  "current_level": 0,
  "level_percent": 0,
  "season_id": 22,
  "prev_seasons_cam_games_played": 0,
  "prev_seasons_cam_discos": 0,
  "latest_season_cam_games_played": 3,
  "latest_season_cam_discos": 0,
  "curr_season_cam_games_played": 3,
  "curr_season_cam_discos": 0,
  "prev_seasons_cam_cs_games_played": 0,
  "prev_seasons_cam_cs_discos": 0,
  "latest_season_cam_cs_games_played": 3,
  "latest_season_cam_cs_discos": 0,
  "curr_season_cam_cs_games_played": 3,
  "curr_season_cam_cs_discos": 0,
  "con_reward": {
    "old_lvl": 0,
    "curr_lvl": 0,
    "next_lvl": 0,
    "require_rank": 0,
    "need_more_play": 0,
    "percentage_before": "0.00",
    "percentage": "0.00"
  },
  "vested_threshold": 5,
  "0": true
}